Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -69,6 +69,12 @@ private DiagnosticsSettings ReadSettings()
                 {
                     string fileContent = null;
                     OperationManager.Attempt(() => fileContent = FileSystemHelpers.ReadAllTextFromFile(_path));
+                    if (DiagnosticsSettings.AutoCorrect(_tracer, fileContent, out fileContent))
+                    {
+                        // if there is corrupted value, reset it back to file system
+                        FileSystemHelpers.WriteAllTextToFile(_path, fileContent);
+                    }
+
                     return JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent);
```

Comment:
Nope! that is why i put in the validation.

![image](https://cloud.githubusercontent.com/assets/79823/11005287/cba22792-8471-11e5-9a2e-d8ee8461d1e7.png)